app/src/main/java/org/thoughtcrime/securesms/crypto/ModernDecryptingPartInputStream.java

```diff
@@ -0,0 +1,84 @@
+package org.thoughtcrime.securesms.crypto;
+
+
+import androidx.annotation.NonNull;
+
+import org.signal.core.util.Conversions;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+
+import javax.crypto.Cipher;
+import javax.crypto.CipherInputStream;
+import javax.crypto.Mac;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+
+public class ModernDecryptingPartInputStream {
+
+  public static InputStream createFor(@NonNull AttachmentSecret attachmentSecret, @NonNull byte[] random, @NonNull File file, long offset)
+      throws IOException
+  {
+    return createFor(attachmentSecret, random, new FileInputStream(file), offset);
+  }
+
+  public static InputStream createFor(@NonNull AttachmentSecret attachmentSecret, @NonNull File file, long offset)
+      throws IOException
+  {
+    FileInputStream inputStream = new FileInputStream(file);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code might contain a resource leak. A resource leak can slow down or crash your system.

**Fix**
Consider closing the following resource in a try-finally block: *inputStream*. Currently, there are execution paths that do not contain closure statements, for example, when ModernDecryptingPartInputStream.createFor throws an exception.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).